numpy/tests/test_financial_rate.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import division, absolute_import, print_function
```

Comment:
I think these tests probably belong in `test_financial.py`. In particular, your `test_rate(self)` function looks like it has been copied from the file.

It probably would make sense to introduce a `TestFinancialRate` in `test_financial.py`, rather than just adding everything to `TestFinancial`.

Reply:
Thanks @eric-wieser  for review. I have added the testcase in existing "test_financial.py" by creating one new method in "TestFinancial" class named "test_rate_nan()".

Reply:
Please do let me know if you need anything in addition to it.